packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -337,6 +340,12 @@ export const UploadControls = (): JSX.Element => {
         Add files
       </ButtonElement>
       <Destination>{history[history.length - 1].prefix}</Destination>
+      <Overwrite
+        defaultChecked={overwrite}
+        handleChange={() => {
+          setOverwrite((overwrite) => !overwrite);
```

Comment:
NIt: To help reduce complexity for future reviewers, avoid reuse of variable names already used in outer scope:
```suggestion
          setOverwrite((prevOverwrite) => !prevOverwrite);
```